src/odemis/acq/milling/plotting.py

```diff
@@ -0,0 +1,150 @@
+"""
+@author: Patrick Cleeve
+
+Copyright © 2025 Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License version 2 as published by the Free
+Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+
+
+### Purpose ###
+
+This module contains plotting utils related to milling tasks.
+
+"""
+from typing import Dict, List
+
+import matplotlib.patches as mpatches
+import matplotlib.pyplot as plt
+from odemis import model
+from odemis.acq.milling.patterns import (
+    MicroexpansionPatternParameters,
+    RectanglePatternParameters,
+    TrenchPatternParameters,
+)
+from odemis.acq.milling.tasks import MillingTaskSettings
+
+# milling pattern colours
+COLOURS = [
+    "yellow","cyan", "magenta", "lime",
+    "orange","hotpink", "green", "blue",
+    "red", "purple",
+]
+
+def _draw_trench_pattern(image: model.DataArray, params: TrenchPatternParameters, colour: str = "yellow", name: str = "Task") -> List[mpatches.Rectangle]:
+    # get parameters
+    width = params.width.value
+    height = params.height.value
+    spacing = params.spacing.value
+    mx, my = params.center.value
+
+    # position in metres from image centre
+    pixel_size = image.metadata[model.MD_PIXEL_SIZE][0] # assume isotropic
+    pmx, pmy = mx / pixel_size, my / pixel_size
+
+    # convert to image coordinates
+    cy, cx = image.shape[0] // 2, image.shape[1] // 2
```

Comment:
Why using `//` (ie, integer division)? From the code after it seems that floats are accepted fine.